pkg/commands/artifact/k8s.go

```diff
@@ -0,0 +1,274 @@
+
+	// Skip downloading vulnerability DB
+	opt.SkipDBUpdate = true
+
+	return initScannerConfig(ctx, opt, cacheClient)
+}
+
+func k8sScanConfig(ctx context.Context, config ScannerConfig, opts types.ScanOptions, a *artifacts.Artifact) (types.Report, error) {
+	fileName, err := createTempFile(a)
+	if err != nil {
+		return types.Report{}, xerrors.Errorf("scan error: %w", err)
+	}
+	defer removeFile(fileName)
+
+	report, err := k8sScan(ctx, fileName, filesystemStandaloneScanner, config, opts)
+	if err != nil {
+		return types.Report{}, xerrors.Errorf("scan error: %w", err)
+	}
+
+	return report, nil
+}
+
+func k8sScan(ctx context.Context, target string, initializeScanner InitializeScanner, config ScannerConfig, opts types.ScanOptions) (types.Report, error) {
+	config.Target = target
+	s, cleanup, err := initializeScanner(ctx, config)
+	if err != nil {
+		log.Logger.Debugf("unexpected error during scanning %s: %s", config.Target, err)
+		return types.Report{}, err
+	}
+	defer cleanup()
+
+	report, err := s.ScanArtifact(ctx, opts)
+	if err != nil {
+		return types.Report{}, xerrors.Errorf("artifact scan failed: %w", err)
+	}
+	return report, nil
+}
+
+func createTempFile(artifact *artifacts.Artifact) (string, error) {
+	filename := fmt.Sprintf("%s-%s-%s-*.yaml", artifact.Namespace, artifact.Kind, artifact.Name)
+
+	file, err := os.CreateTemp("", filename)
+	if err != nil {
+		return "", xerrors.Errorf("creating tmp file error: %w", err)
+	}
+	defer func() {
+		if err := file.Close(); err != nil {
+			log.Logger.Errorf("failed to close temp file %s: %s:", file.Name(), err)
+		}
+	}()
+
+	// TODO(josedonizetti): marshal and return as byte slice should be on the trivy-kubernetes library?
+	data, err := yaml.Marshal(artifact.RawResource)
+	if err != nil {
+		removeFile(filename)
+		return "", xerrors.Errorf("marshaling resource error: %w", err)
+	}
+
+	_, err = file.Write(data)
+	if err != nil {
+		removeFile(filename)
+		return "", xerrors.Errorf("writing tmp file error: %w", err)
+	}
+
+	return file.Name(), nil
+}
+
+func newK8sResource(artifact *artifacts.Artifact, report types.Report, err error) k8sReport.Resource {
+	results := make([]types.Result, 0, len(report.Results))
+	// fix target name
+	for _, result := range report.Results {
+		// if resource is a kubernetes file fix the target name,
+		// to avoid showing the temp file that was removed.
+		if result.Type == "kubernetes" {
+			result.Target = fmt.Sprintf("%s/%s", artifact.Kind, artifact.Name)
+		}
+		results = append(results, result)
+	}
+
+	k8sreport := k8sReport.Resource{
+		Namespace: artifact.Namespace,
+		Kind:      artifact.Kind,
+		Name:      artifact.Name,
+		Results:   results,
+	}
+
+	// if there was any error during the scan
+	if err != nil {
+		k8sreport.Error = err.Error()
+	}
+
+	return k8sreport
+}
+
+func removeFile(filename string) {
+	if err := os.Remove(filename); err != nil {
+		log.Logger.Errorf("failed to remove temp file %s:%w:", filename, err)
```

Comment:
Looks like this issue is remaining.
https://github.com/aquasecurity/trivy/pull/2065#discussion_r869539653